tests/integrationv2/test_fragmentation.py

```diff
@@ -55,13 +62,25 @@ def test_s2n_server_low_latency(managed_process, multi_cipher, provider, protoco
         assert random_bytes in results.stdout
 
 
-@pytest.mark.uncollect_if(func=invalid_test_parameters)
-@pytest.mark.parametrize("multi_cipher", [multi_cipher], ids=multi_cipher_name)
-@pytest.mark.parametrize("provider", [OpenSSL])
+def invalid_test_parameters_frag_len(*args, **kwargs):
+    provider = kwargs.get("provider")
+    frag_len = kwargs.get("frag_len")
+
+    # check to make sure frag_len is compatible with gnutls
+    if provider == GnuTLS:
+        if frag_len > 4096:
+            return True
+
+    return invalid_test_parameters(*args, **kwargs)
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters_frag_len)
+@pytest.mark.parametrize("cipher", CIPHERS_TO_TEST, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL, GnuTLS])
 @pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
-@pytest.mark.parametrize("certificate", [Certificates.RSA_4096_SHA384, Certificates.ECDSA_384], ids=get_parameter_name)
-@pytest.mark.parametrize("frag_len", [512, 2048, 8192, 12345, 16384], ids=get_parameter_name)
```

Comment:
We're checking frag_len, but it looks like everything above 2048 got dropped? Was that intentional?

Reply:
This was a mistake, thanks! 7c15b2d27d38458ba3eaa6516e0eadbc450a61e6